controllers/cloud.redhat.com/makers/maker.go

```diff
@@ -191,6 +200,201 @@ func applyPodAntiAffinity(t *core.PodTemplateSpec) {
+	labels := app.GetLabels()
+	labels["pod"] = nn.Name
+	app.SetObjectMeta(jc, crd.Name(nn.Name), crd.Labels(labels))
+
+	pod := job.PodSpec
+
+	jc.Spec.Schedule = job.Schedule
+
+	jc.Spec.JobTemplate.ObjectMeta.Labels = labels
+	jc.Spec.JobTemplate.Spec.Template.ObjectMeta.Labels = labels
+	jc.Spec.JobTemplate.Spec.Template.Spec.RestartPolicy = core.RestartPolicyNever
+
+	jc.Spec.JobTemplate.Spec.Template.Spec.ImagePullSecrets = []core.LocalObjectReference{
+		{Name: "quay-cloudservices-pull"},
+	}
+
+	envvar := pod.Env
+	envvar = append(envvar, core.EnvVar{Name: "ACG_CONFIG", Value: "/cdapp/cdappconfig.json"})
+
+	var livenessProbe core.Probe
+	var readinessProbe core.Probe
+
+	if pod.LivenessProbe != nil {
+		livenessProbe = *pod.LivenessProbe
+	} else {
+		livenessProbe = core.Probe{}
+	}
+	if pod.ReadinessProbe != nil {
+		readinessProbe = *pod.ReadinessProbe
+	} else {
+		readinessProbe = core.Probe{}
+	}
+
+	var lcpu, lmemory, rcpu, rmemory resource.Quantity
+	nullCPU := resource.Quantity{Format: resource.DecimalSI}
+	nullMemory := resource.Quantity{Format: resource.BinarySI}
+
+	if *pod.Resources.Limits.Cpu() != nullCPU {
+		lcpu = pod.Resources.Limits["cpu"]
+	} else {
+		lcpu = env.Spec.ResourceDefaults.Limits["cpu"]
+	}
+
+	if *pod.Resources.Limits.Memory() != nullMemory {
+		lmemory = pod.Resources.Limits["memory"]
+	} else {
+		lmemory = env.Spec.ResourceDefaults.Limits["memory"]
+	}
+
+	if *pod.Resources.Requests.Cpu() != nullCPU {
+		rcpu = pod.Resources.Requests["cpu"]
+	} else {
+		rcpu = env.Spec.ResourceDefaults.Requests["cpu"]
+	}
+
+	if *pod.Resources.Requests.Memory() != nullMemory {
+		rmemory = pod.Resources.Requests["memory"]
+	} else {
+		rmemory = env.Spec.ResourceDefaults.Requests["memory"]
+	}
+
+	resources := core.ResourceRequirements{
+		Limits: core.ResourceList{
+			"cpu":    lcpu,
+			"memory": lmemory,
+		},
+		Requests: core.ResourceList{
+			"cpu":    rcpu,
+			"memory": rmemory,
+		},
+	}
+
+	c := core.Container{
+		Name:         nn.Name,
+		Image:        pod.Image,
+		Command:      pod.Command,
+		Args:         pod.Args,
+		Env:          envvar,
+		Resources:    resources,
+		VolumeMounts: pod.VolumeMounts,
+		Ports: []core.ContainerPort{{
+			Name:          "metrics",
+			ContainerPort: env.Spec.Providers.Metrics.Port,
+		}},
+		ImagePullPolicy: core.PullIfNotPresent,
+	}
+
+	if (core.Probe{}) != livenessProbe {
+		c.LivenessProbe = &livenessProbe
+	}
+	if (core.Probe{}) != readinessProbe {
+		c.ReadinessProbe = &readinessProbe
+	}
+
+	c.VolumeMounts = append(c.VolumeMounts, core.VolumeMount{
+		Name:      "config-secret",
+		MountPath: "/cdapp/",
+	})
+
+	jc.Spec.JobTemplate.Spec.Template.Spec.Containers = []core.Container{c}
+
+	if len(pod.InitContainers) > 0 {
+		jc.Spec.JobTemplate.Spec.Template.Spec.InitContainers = make([]core.Container, len(pod.InitContainers))
+	}
+
+	for i, ic := range pod.InitContainers {
```

Comment:
Similar comment here.  Could this be put into a common function to be used both here and `initDeployment`?

Reply:
I'll DRY it up